google-cloud-logging/lib/google/cloud/logging/rails.rb

```diff
@@ -64,60 +49,96 @@ class Railtie < ::Rails::Railtie
-
-            project_id = logging_config[:project_id]
-            keyfile = logging_config[:keyfile]
-            resource_type = logging_config[:resource_type]
-            resource_labels = logging_config[:resource_labels]
-            log_name = logging_config[:log_name]
-
-            logging = Google::Cloud::Logging.new project: project_id,
-                                                 keyfile: keyfile
-            resource =
-              Logging::Middleware.build_monitored_resource resource_type,
-                                                           resource_labels
-
-            app.config.logger = logging.logger log_name, resource
-            app.middleware.insert_before Rails::Rack::Logger,
-                                         Google::Cloud::Logging::Middleware,
-                                         logger: app.config.logger
-          end
+          self.class.consolidate_rails_config app.config
+
+          self.class.init_middleware app if Cloud.configure.use_logging
         end
 
         ##
-        # Determine whether to use Stackdriver Logging or not.
-        #
-        # Returns `true` if Stackdriver Logging is enabled for this application.
-        # That is, if all of the following are true:
-        #
-        # * A valid GCP `project_id` is available, either because the
-        #   application is hosted on Google Cloud or because it is set in the
-        #   configuration.
-        # * The API is able to authenticate, again either because the
-        #   application is hosted on Google Cloud or because an appropriate
-        #   keyfile is provided in the configuration.
-        # * Either the Rails environment is set to `production` or the
-        #   `config.google_cloud.use_logging` configuration is explicitly set to
-        #   `true`.
+        # @private Init Logging integration for Rails. Setup configuration and
+        # insert the Middleware.
+        def self.init_middleware app
+          project_id = Logging.configure.project_id
+          keyfile = Logging.configure.keyfile
+          resource_type = Logging.configure.monitored_resource.type
+          resource_labels = Logging.configure.monitored_resource.labels
+          log_name = Logging.configure.log_name
+
+          logging = Google::Cloud::Logging.new project: project_id,
+                                               keyfile: keyfile
+          resource =
+            Logging::Middleware.build_monitored_resource resource_type,
+                                                         resource_labels
+
+          app.config.logger = logging.logger log_name, resource
+          app.middleware.insert_before Rails::Rack::Logger,
+                                       Google::Cloud::Logging::Middleware,
+                                       logger: app.config.logger,
+                                       project_id: project_id,
+                                       keyfile: keyfile
```

Comment:
Is all this even necessary now? Wouldn't the defaults from the current config be correct?